### PR TITLE
fix(replays): Cap the number of aggregation targets we'll search for to match the number of aggregations we'll return

### PR DESCRIPTION
**Pull request by cmanallen**
Sometimes the client will send more aggregation keys than the endpoint is capable of returning. We should handle these situations gracefully. The proposed solution is to cap the list and silently drop extra aggregation keys which matches current production behavior without the wild memory usage.

Addresses: https://sentry.sentry.io/issues/4802777330

### CODE REVIEW

File: src/sentry/replays/usecases/replay_counts.py [hunk 1]
```diff
@@ -56,6 +56,13 @@ def _get_replay_id_mappings(
         # if we want to validate list of replay_ids existence
         return {v: [v] for v in value}
 
+    # Any number of issues can be specified however we're only going to return at
+    # most 25 of them. Cap the lists length to 25. The other ids are silently
+    # dropped.
+    #
+    # NOTE: Should this be an error?
```
Comment (bruno-garcia) [inline src/sentry/replays/usecases/replay_counts.py:63]: Could we keep tabs on this through metrics? if we see a jump in calls with over 25 we can get eyes on it

**Review by bruno-garcia — APPROVED** [review]

File: src/sentry/replays/usecases/replay_counts.py (hunk 1 above)
Comment (bruno-garcia) [reply src/sentry/replays/usecases/replay_counts.py:63]: Definitely silently ignoring might be a problem in the future to figure out what's going on

Comment (cmanallen) [reply src/sentry/replays/usecases/replay_counts.py]: Adding a logger.warning which will appear in Sentry.

**Review by JoshFerge — COMMENTED** [review]
we already have `MAX_VALS_PROVIDED`, why is a new abstraction needed?

**Comment by cmanallen** [comment]
@JoshFerge Didn't realize that was there.  Refer to the Sentry issue in the pull description.  >65 issue ids are being queried.  Something is wrong with that method.  We can resolve in a follow-up.

**Comment by cmanallen** [comment]
@JoshFerge

> I also don't understand why including more IDs results in high memory usage -- could you explain this?

Each aggregation state consumes memory.  I have no idea how many ids are being passed to the query but its more than 25.

**Comment by cmanallen** [comment]
> I think because I don't prefer silently returning incorrect data, I would revert this PR as is, but that's just me.

This is what we're doing right now.  We're limiting by 25 but more than 25 ids are being passed.  No change from current production behavior so no need to revert.

**Review by JoshFerge — COMMENTED** [review]
also, can we please add a test for this behavior? // there should also be a test that is already testing the number of ids passed in.

**Comment by cmanallen** [comment]
@JoshFerge I see my mistake.  I misinterpreted a filter on the project-id column for a filter on the issue-id column.  So that validation you mentioned is working.  I reverted the PR.  This change should not cause any issues.  Because the validation is working and this new validation matches it, it should be a no-op.

**Pull request merged**